sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +379,482 @@ def transform(self, X):
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
+            self,
+            missing_values='NaN',
+            imputation_order='monotone',
+            n_imputations=100,
+            n_burn_in=10,
+            estimator=None,
+            n_nearest_features=None,
+            initial_strategy="mean",
+            min_value=None,
+            max_value=None,
+            verbose=False,
+            random_state=None):
+
+        self.missing_values = missing_values
+        self.imputation_order = imputation_order
+        self.n_imputations = n_imputations
+        self.n_burn_in = n_burn_in
+        self.estimator = estimator
+        self.n_nearest_features = n_nearest_features
+        self.initial_strategy = initial_strategy
+        self.min_value = min_value
+        self.max_value = max_value
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _impute_one_feature(self,
+                            X_filled,
+                            mask_missing_values,
+                            feat_idx,
+                            neighbor_feat_idx,
+                            estimator=None,
+                            min_std=1e-6):
+        """Imputes a single feature from the others provided.
+
+        This function predicts the missing values of one of the features using
+        the current estimates of all the other features. The `estimator` must
+        support `return_std=True` in its ``predict`` method for this function
+        to work.
+
+        Parameters
+        ----------
+        X_filled : array-like
+            Input data with the most recent imputations.
+
+        mask_missing_values : array-like
+            Input data's missing indicator matrix.
+
+        feat_idx : integer
+            Index of the feature currently being imputed.
+
+        neighbor_feat_idx : array-like
+            Indices of the features to be used in imputing `feat_idx`.
+
+        estimator : object, default=BayesianRidgeRegression
+            The estimator to use at this step of the round-robin imputation.
+            As of now, it must support `return_std` in its ``predict`` method.
+            If None, it will be fit, otherwise not.
+
+        min_std : float, optional (default=1e-5)
+            The smallest allowable standard deviation for the posterior
+            sampling step.
+
+        Returns
+        -------
+        X_filled : array-like
+            Input data with `X_filled[missing_row_mask, feat_idx]`
+            updated.
+
+        estimator : estimator with sklearn API
+            The fitted estimator used to impute
+            `X_filled[missing_row_mask, feat_idx]`.
+        """
+
+        # if nothing is missing, just return the default
+        rng = check_random_state(self.random_state)
```

Comment:
Surely you don't want this kind of correlation from feature to feature... You need to check_random_state once at the beginning of imputation.

Reply:
I am not familiar with the proper etiquette here. Later on in this function we have `imputed_values = rng.normal(loc=mus, scale=np.maximum(sigmas, min_std))`. Does that mean `rng` should be made `self.rng` at the beginning of `fit_transform`?